streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java

```diff
@@ -287,7 +279,7 @@ public void shouldGetExistingThreadLevelSensor() {
 
         final Sensor actualSensor = streamsMetrics.threadLevelSensor(THREAD_ID1, SENSOR_NAME_1, recordingLevel);
 
-        verify(metrics);
+        verify(metrics).getSensor("internal.test-thread-1.s.sensor1");
```

Comment:
`shouldGetExistingThreadLevelSensor` test invokes `setupGetExistingSensorTest(metrics)` which does the following:
```java
    private void setupGetExistingSensorTest(final Metrics metrics) {
        when(metrics.getSensor(anyString())).thenReturn(sensor);
    }
```

An explicit verify is added to be sure `metrics.getSensor()` is invoked with the exact string, and not `anyString()` as stated in stub in `setupGetExistingSensorTest` method.

The same reasoning applies to other explicit `verify(metrics).getSensor(...)` calls.

Reply:
Good reasoning!

Why not pass the sensor name to `setupGetExistingSensorTest()` and remove the explicit `verify()`?
Please use `INTERNAL_PREFIX + SENSOR_PREFIX_DELIMITER + THREAD_ID1 + SENSOR_NAME_DELIMITER + ...` for the sensor names.  

From a migration point of view the verification is not strictly needed. Maybe you want to do the additional verifications in a separate PR and there you could also add the verifications for `setupGetNewSensorTest()` by using the captured names. Maybe you can even get rid of the capture thingy there since we only use it to verify if the sensor name is the same in both calls which could probably also be done by passing the sensor name to `setupGetNewSensorTest()`.

All of this also applies to the other verifications you added.